app/models/user_api.rb

```diff
@@ -0,0 +1,27 @@
+class UserApi < ApplicationRecord
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, presence: true
+  validates :url, presence: true
+  validates :redirect_url, presence: true
+
+  belongs_to :user
+
+  def overwrite_key
+    update(key: UserApi.generate_key)
+  end
+
+  def update(params)
```

Comment:
@rdavid1099 Probably use `before_validation` instead of monkey patching `update`. Although, I like to see an understanding of `super` 👍 . Even better IMO would be to build a `UserApiCreator` and use straight Ruby to update the object in memory before persisting to the DB.